projects/charger/src/charger_pin.c

```diff
@@ -1,49 +1,46 @@
 #include "charger_pin.h"
 
 #include <stddef.h>
+#include <stdbool.h
 
+#include "adc.h"
 #include "charger_events.h"
 #include "event_queue.h"
 #include "gpio.h"
 #include "gpio_it.h"
 #include "interrupt_def.h"
+#include "soft_timer.h"
 #include "status.h"
 
-static void prv_charger_pin_it(const GPIOAddress *address, void *context) {
-  (void)context;
-  GPIOState state = NUM_GPIO_STATES;
-  gpio_get_state(address, &state);
-  switch (state) {
-    // TODO(ELEC-355): Determine if this logic is correct.
-    case GPIO_STATE_LOW:
-      event_raise(CHARGER_EVENT_DISCONNECTED, 0);
-      break;
-    case GPIO_STATE_HIGH:
-      event_raise(CHARGER_EVENT_CONNECTED, 0);
-    case NUM_GPIO_STATES:
-    default:
-      // Unreachable
-      break;
+#define CHARGER_PIN_CONNECTED_THRESHOLD 1700
```

Comment:
Documentation on where this came from would be nice (also units)

This might belong better in the header?

Reply:
Done.